File: py/money.py
class Money:
    def __init__(self, amount, currency):
        self.amount = amount
        self.currency = currency

    def times(self, multiplier):
        return Money(self.amount * multiplier, self.currency)

    def divide(self, divisor):
        return Money(self.amount / divisor, self.currency)

    def __eq__(self, other):
        return self.amount == other.amount and self.currency == other.currency

    def __str__(self):
        return f"{self.currency} {self.amount:0.2f}"

    def __add__(self, a):
        if a is not None and self.currency == a.currency:
            return Money(self.amount + a.amount, self.currency)
        else:
            return None

File: js/money.js
class Money {
    constructor(amount, currency) {
        this.amount = amount;
        this.currency = currency;
    }

    times(multiplier) {
        return new Money(this.amount * multiplier, this.currency);
    }

    divide(divisor) {
        return new Money(this.amount / divisor, this.currency);
    }

    add(money) {
        if (money.currency !== this.currency) {
            throw new Error("Cannot add " + money.currency + " to " + this.currency);
        }
        return new Money(this.amount + money.amount, this.currency);
    }

}

module.exports = Money;
